### ISSUE TITLE
Problems to detect small note heads resp. stems ending in small note heads

### ISSUE THREAD

**Issue by Bacchushlg** [Audiveris/audiveris#306]
![Run to You (2)](https://user-images.githubusercontent.com/33522630/66252216-f7ad9f80-e758-11e9-89e2-3571c5ac8414.jpg)

This score contains a couple of small note heads.
Initially I intended to ignore them (and just add in the notation program), but it seems that Audiveris has problems with stems that do not end in a note head (makes sense!).
So I decided to activate the recognition of small note heads (full and open). Unfortunately this does not help a lot. Still most of the small heads stay unrecognized.

**Comment by oleedd** [comment]
But what do such notes mean there? Optional notes?

**Comment by Bacchushlg** [comment]
small note heads are mostly used in case of a score with two or more verses, in case that the verses are slightly different in rhythm (syllables). 
In the score above it is meant for voice splitting: the bottom not lines is combined (tenor and bass), and the bass line is partly split (bass 1 and bass 2).

**Comment by oleedd** [comment]
Does music theory allow this (any source)? It is grace notes.

**Comment by Bacchushlg** [comment]
no, grace notes are something completely different. They are - in theory - not considered in the rhythm of a measure.
These small note heads are a real part of the score - in a way they can be seen as optional notes. In the case above it means: basic level choirs will ignore them, more sophisticated one's will split the voices as proposed.

**Comment by oleedd** [comment]
Hm. Interesting.

> no, grace notes are something completely different. They are - in theory - not considered in the rhythm of a measure.

I mean that small notes are grace notes normally. So it just nonstandard usage, not documented in the official rules. It sometimes happens in music. Optional notes (or a dubbed note in another staff) should be written in brackets.

**Comment by Bacchushlg** [comment]
I have dozns if scores that use smalls notes in this way. So, let's say that it is good practice to use them this way (and it really good readable this way).

Am 13. Oktober 2019 14:24:09 MESZ schrieb oleedd <notifications@github.com>:

**Comment by hekkup** [comment]
They are called cue notes and they are a standard way to show what another instrument or voice is doing, making it easier to follow the piece. For example you can have clarinet part with cue notes showing the melody piano accompaniment part, so you will have a hint where the piece is going. Moreover, the small notes are used for example in runs which have a great many of notes which would take a lot of space to print in normal size. For example, piano pieces of Chopin and Liszt have these.
